### PR TITLE
Fix `ClassCastException` in `GraphqlService`

### LINKED ISSUE #4104: Graphql: Return 400 instead of 500 when client sends incorrect request in json
Currently Armeria returns a 500 error when a client sends in either invalid json or invalid graphql request.

I agree with what @jrhee17 wrote in https://github.com/line/armeria/pull/4069#discussion_r801222068 , the server should return 400.

### PR DESCRIPTION
**Pull request by heowc**
Motivation:

In GraphqlService, if `query` and `operationName` are not of string
type, `ClassCastException` is thrown

Modifications:

- Add the validation code that checks the type before implicit conversion.

Result:

- No `ClassCastException` is thrown.
- Fixes #4104

### CODE REVIEW

File: graphql-protocol/src/main/java/com/linecorp/armeria/server/graphql/protocol/AbstractGraphqlService.java [hunk 1]
```diff
@@ -164,6 +164,19 @@ protected abstract HttpResponse executeGraphql(ServiceRequestContext ctx, Graphq
...
+    @Nullable
+    private static String toStringFromJson(@Nullable Object maybeString) {
+        if (maybeString == null) {
+            return null;
+        }
+
+        if (maybeString instanceof String) {
+            return (String) maybeString;
+        } else {
+            throw new IllegalArgumentException("Unknown parameter type variables");
```
Comment (ikhoon) [inline graphql-protocol/src/main/java/com/linecorp/armeria/server/graphql/protocol/AbstractGraphqlService.java:176]: Could you check the returned status? 400 bad request seems good for the invalid `operationName` but I guess this response status is 500.

**Comment by ikhoon** [comment]
> I believe with [0aa7c6f](https://github.com/line/armeria/commit/0aa7c6fbbe8b70587f545f12f6e5782f3305ab67), #4104 is also resolved 👍

I have forgotten the issue. Thanks for reminding me.

**Review by jrhee17 — APPROVED** [review]
Looks great! Thanks @heowc and @ikhoon 🙏 👍 🙇

**Review by trustin — APPROVED** [review]
Thanks, @heowc! 🙇

**Review by ikhoon — APPROVED** [review]
Thanks, @heowc!

**Pull request merged**